src/repl.rs

```diff
@@ -76,25 +82,55 @@ fn find_info(args: WithVM<RootStr>) -> IO<Result<String, String>> {
     IO::Value(Ok(buffer))
 }
 
-fn compile_repl(vm: &Thread) -> Result<(), Box<StdError + Send + Sync>> {
-    fn input(prompt: &str) -> IO<Option<String>> {
-        let input = ::linenoise::input(prompt);
+struct Editor(Mutex<::rustyline::Editor<()>>);
 
-        if let Some(ref input) = input {
-            if !input.trim().is_empty() {
-                ::linenoise::history_add(input);
-            }
-        }
+impl fmt::Debug for Editor {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "File")
```

Comment:
Copy paste mistake, will fix. The implementation is only there to satisfy the userdata trait which requires it.